dynamiqs/mesolve/rouchon.py

```diff
@@ -75,14 +44,21 @@ def forward(self, t: float, rho: Tensor) -> Tensor:
         # rho: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
 
         # compute cached operators
-        # H, Hnh, M0: (b_H, 1, n, n)
+        # H, Hnh, M0, S, T: (b_H, 1, n, n)
         H = self.H(t)
         Hnh = self.Hnh(H)
         M0 = self.M0(Hnh)
 
         # compute rho(t+dt)
+        if self.options.cholesky_normalization:
+            T = self.T(M0)
+            rho = torch.linalg.solve_triangular(T.mH, rho, upper=True)
+            rho = torch.linalg.solve_triangular(T, rho, upper=False, left=False)
+
         rho = kraus_map(rho, M0) + kraus_map(rho, self.M1s)  # (b_H, b_rho, n, n)
-        rho = unit(rho)
+
+        if not self.options.cholesky_normalization:
+            rho = unit(rho)
```

Comment:
For now I'd like to keep it because we should return a normalized quantum state. However, we should indeed keep track of the trace correction at every time step, and return a warning if it diverges (and return by how much it diverged in `Result`). If it's ok with you, I'll implement this in a separate PR.